### ISSUE TITLE
Electron support - possible solution?

### ISSUE THREAD

**Issue by Cleudi** [flotwig/cypress-log-to-output#2]
There is an env named ELECTRON_ENABLE_LOGGING=1 which seems to fix the electron issue. Can you please test this?

See also https://github.com/cypress-io/cypress/issues/3199#issuecomment-492728331

**Comment by konekoya** [comment]
> There is an env named ELECTRON_ENABLE_LOGGING=1 which seems to fix the electron issue. Can you please test this?

I tried this today. But it didn't work. But it works okay with @xhubbsaxo's snippet

**Comment by flotwig** [comment]
You can also run Cypress with [debug logs enabled][debug] to get the Electron logs in `stderr`:

* On Linux/macOS: `DEBUG=cypress:server:browsers:electron cypress ...`
* On Windows: `npx cross-env DEBUG=cypress:server:browsers:electron cypress ...`

It will share some extra data, but console.logs and stuff are in there too.

[debug]: https://docs.cypress.io/guides/guides/debugging.html#Print-DEBUG-logs

**Comment by nmschulte-aviture** [comment]
Does the new browser launch API, with Cypress v4, make it possible to support Electron?  https://docs.cypress.io/api/plugins/browser-launch-api.html#Usage

**Comment by flotwig** [comment]
unfortunately not @nmschulte-aviture

**Comment by pkyeck** [comment]
> You can also run Cypress with [debug logs enabled](https://docs.cypress.io/guides/guides/debugging.html#Print-DEBUG-logs) to get the Electron logs in `stderr`:
> 
> * On Linux/macOS: `DEBUG=cypress:server:browsers:electron cypress ...`
> * On Windows: `npx cross-env DEBUG=cypress:server:browsers:electron cypress ...`
> 
> It will share some extra data, but console.logs and stuff are in there too.

That works – thanks.
But do you have an idea how we could filter + reformat the DEBUG-logs?

```
cypress:server:browsers:electron console message: { column: 1430321, level: 'log', line: 2, source: 'console-api', text: ' prev state color: #9E9E9E; font-weight: bold [object Object]', url: 'http://ui/static/js/2.feecd219.chunk.js' }
cypress:server:browsers:electron console message: { column: 1430441, level: 'log', line: 2, source: 'console-api', text: ' action     color: #03A9F4; font-weight: bold [object Object]', url: 'http://ui/static/js/2.feecd219.chunk.js' }
```

These super-noisy lines aren't helpful after all :(

**Comment by flotwig** [comment]
@pkyeck no possible way to do it currently. Maybe you could hack something by piping through awk/sed

**Comment by pkyeck** [comment]
@flotwig yeah, already thought about that. But it's too late at this point because we have a lot of `[object Object]`s in there that should be `stringified` (or something similar) before printing them to stdout.
